### PR TITLE
Marika

### PR DESCRIPTION
**Pull request by dmitriigo**

### CODE REVIEW

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 1]
```diff
@@ -1,38 +1,76 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_helpees", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "helpee_id")})
+    private List<Helpee> helpees = new ArrayList<>();
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_candidates", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "candidate_id")})
+    private List<Candidate> candidates = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Customer.java:50]: Same problem ^. we have only Customer. We don't have entity Candidate

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 2]
```diff
@@ -1,38 +1,76 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_helpers", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "helper_id")})
+    private List<Helper> helpers = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Customer.java:55]: Same problem ^

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 3]
```diff
@@ -1,38 +1,76 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_ratings", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "rating_id")})
+    private List<HelpQuestRating> helpQuestRatings = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Customer.java:60]: Same problem ^. Entity name: Rating. Not HelpQuestRating

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 4]
```diff
@@ -1,38 +1,76 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "customer_ratings", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "rating_id")})
+    private List<CustomerRating> customerRatings = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Customer.java:65]: Same problem ^.

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 5]
```diff
@@ -1,38 +1,76 @@
...
+
+    @OneToMany(fetch = FetchType.EAGER)
+    private List<CandidateRequest> candidateRequests = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Customer.java:68]: Same problem ^.

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 6]
```diff
@@ -1,38 +1,76 @@
...
-    @OneToMany(cascade=CascadeType.ALL,mappedBy = "ratingReceiver")
-    private List<Rating> receivedRatings;
+    @OneToMany(fetch = FetchType.EAGER)
+    private List<HelpQuest> helpQuests = new ArrayList<>();
 
-    @OneToMany(cascade=CascadeType.ALL, mappedBy = "ratingAuthor")
-    private List<Rating> givenRatings;
-}
+    /*public Customer() {
+        Role = Role.CUSTOMER;
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Customer.java:74]: What is Role? Role is type of some variable. name of variable is "role".
change this string to "this.role = Role.Customer"

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 7]
```diff
@@ -1,44 +1,66 @@
...
+import lombok.experimental.Helper;
 import javax.persistence.*;
 import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.List;
 
 @Data
 @Entity
-public class HelpQuest {
+public class HelpQuest<Helpee, Candidate, CustomerRating, CandidateRequest, HelpQuestRating> {
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:12]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 8]
```diff
@@ -1,44 +1,66 @@
...
\ No newline at end of file
+
+    @Column(length = 350, nullable = false)
+    private String requestMessage;
+
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_helpees", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "helpee_id")})
+    private List<Helpee> helpees = new ArrayList<Helpee>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:39]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 9]
```diff
@@ -1,44 +1,66 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_helpers", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "helper_id")})
+    private List<Helper> helpers = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:44]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 10]
```diff
@@ -1,44 +1,66 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_candidates", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "candidate_id")})
+    private List<Candidate> candidates = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:49]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 11]
```diff
@@ -1,44 +1,66 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "helpQuest_ratings", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "rating_id")})
+    private List<HelpQuestRating> helpQuestRatings = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:54]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 12]
```diff
@@ -1,44 +1,66 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "customer_ratings", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "rating_id")})
+    private List<CustomerRating> customerRatings = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:59]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuest.java [hunk 13]
```diff
@@ -1,44 +1,66 @@
...
+
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "candidate_requests", joinColumns = {@JoinColumn(name = "helpQuest_id")},
+            inverseJoinColumns = {@JoinColumn(name = "request_id")})
+    private List<CandidateRequest> candidateRequests = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuest.java:64]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/HelpQuestStatus.java [hunk 14]
```diff
@@ -1,10 +1,22 @@
 package com.bestgroup.helppy.entities;
 
+
+import javax.persistence.Table;
+
+@Table
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuestStatus.java:6]: Delete annotation in enum

File: src/main/java/com/bestgroup/helppy/entities/HelpQuestStatus.java [hunk 15]
```diff
@@ -1,10 +1,22 @@
...
+    CANCELLED("Cancelled", 4),
+    DELETED("Deleted", 5);
+
+    String name;
+    int id;
+
+    HelpQuestStatus(String name, int id) {
+        this.name = name;
+        this.id = id;
+    }
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuestStatus.java:21]: Add getters and setters

File: src/main/java/com/bestgroup/helppy/entities/HelpQuestType.java [hunk 16]
```diff
@@ -1,6 +1,18 @@
 package com.bestgroup.helppy.entities;
 
+
+import javax.persistence.Table;
+
+@Table
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuestType.java:6]: Delete annotation

File: src/main/java/com/bestgroup/helppy/entities/HelpQuestType.java [hunk 17]
```diff
@@ -1,6 +1,18 @@
...
+    REQUEST("Request", 0),
+    OFFER("Offer", 1);
+
+    String name;
+    int id;
+
+    HelpQuestType(String name, int id) {
+        this.name = name;
+        this.id = id;
+    }
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/HelpQuestType.java:17]: add getters and setters

File: src/main/java/com/bestgroup/helppy/entities/Rating.java [hunk 18]
```diff
@@ -3,28 +3,30 @@
...
 import javax.persistence.*;
-import java.sql.Timestamp;
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
 
 @Entity
 @Data
-public class Rating {
+public class Rating<RatingReceiver, RatingAuthor> {
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Rating.java:12]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/Rating.java [hunk 19]
```diff
@@ -3,28 +3,30 @@
...
     private String Comment;
     private int rating;
-    @ManyToOne
-    @JoinColumn(name = "helpQuest")
-    private HelpQuest helpQuest;
     private LocalDateTime date;
 
+    @ManyToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "rating_receivers")
+    private List<RatingReceiver> ratingReceivers = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Rating.java:26]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/Rating.java [hunk 20]
```diff
@@ -3,28 +3,30 @@
...
+
+    @OneToMany(cascade = CascadeType.ALL,
+            mappedBy = "rating_author", fetch = FetchType.EAGER)
+    private List<RatingAuthor> ratingAuthors = new ArrayList<>();
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Rating.java:30]: Problem with generics^

File: src/main/java/com/bestgroup/helppy/entities/Role.java [hunk 21]
```diff
@@ -1,7 +1,18 @@
 package com.bestgroup.helppy.entities;
 
+import javax.persistence.Table;
+
+@Table
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Role.java:5]: delete @Table

File: src/main/java/com/bestgroup/helppy/entities/Role.java [hunk 22]
```diff
@@ -1,7 +1,18 @@
...
+    CUSTOMER("Customer", 1),
+    DISABLED("Disabled", 2);
+
+    String name;
+    int id;
+
+    Role(String name, int id) {
+        this.name = name;
+        this.id = id;
+    }
```
Comment (dmitriigo) [inline src/main/java/com/bestgroup/helppy/entities/Role.java:17]: add getters and setters

File: src/main/java/com/bestgroup/helppy/entities/Customer.java [hunk 23]
```diff
@@ -1,38 +1,76 @@
...
-    private List<HelpQuest> favoriteHelpQuests;
 
-    @ManyToMany(cascade=CascadeType.ALL)
-    @JoinTable(name="favorite_customers",joinColumns = {@JoinColumn(name="customer_id")}, inverseJoinColumns = {@JoinColumn(name="favorite_customer_id")})
-    private List<Customer> favoriteCustomers;
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+    @JoinTable(name = "favorite_helpQuest",
+            joinColumns = {@JoinColumn(name = "customer_id")},
+            inverseJoinColumns = {@JoinColumn(name = "helpQuest_id")})
+    private List<FavoriteHelpQuest> favoriteHelpQuests = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (MarikaTamberg) [reply src/main/java/com/bestgroup/helppy/entities/Customer.java:34]: FavoriteHelpQuest is on our ERD

**Pull request closed without merging**